identity/test/actions/AuthenticatedActionsTest.scala

```diff
@@ -82,4 +80,75 @@ class AuthenticatedActionsTest extends WordSpecLike with MockitoSugar with Scala
       })
     }
   }
+
+  "The consent journey redirect action" should {
+    def failTest: AuthRequest[AnyContent] => Result = _ => fail("Block was invoked")
+
+    "redirect to /signin when the user is not authenticated" in new TestFixture {
+      val originalUrl = "https://profile.thegulocal.com/consents"
+      val request = Request(FakeRequest("GET", originalUrl), AnyContent())
+
+      when(authService.fullyAuthenticatedUser(any[RequestHeader])).thenReturn(None)
+      when(authService.consentCookieAuthenticatedUser(any[RequestHeader])).thenReturn(None)
+
+      val result = actions.consentsRedirectAction().apply(failTest)(request)
+      val expectedLocation = s"/signin?INTCMP=email&returnUrl=${URLEncoder.encode(originalUrl, "utf-8")}"
+      whenReady(result) { res =>
+        res.header.status shouldBe 303
+        res.header.headers should contain("Location" -> expectedLocation)
+      }
+    }
+
+    "not redirect and return 200 when a user is authenticated via an RP cookie" in new TestFixture {
+      val originalUrl = "https://profile.thegulocal.com/consents"
+      val request = Request(FakeRequest("GET", originalUrl), AnyContent())
+
+      when(authService.fullyAuthenticatedUser(any[RequestHeader])).thenReturn(None)
+      when(authService.consentCookieAuthenticatedUser(any[RequestHeader])).thenReturn(Some(userWithRpCookie))
+      when(client.me(any[Auth])).thenReturn(Future(Right(user)))
+      when(profileRedirectService.toConsentsRedirect(any[User], any[RequestHeader])).thenReturn(Future(NoRedirect))
+
+      val mockFunc = mock[Int => Result]
+      when(mockFunc.apply(1)) thenReturn mock[Result]
+      def callMock: AuthRequest[AnyContent] => Result = _ => mockFunc.apply(1)
+
+      val result = actions.consentsRedirectAction().apply(callMock)(request)
+      whenReady(result) { res =>
+        verify(mockFunc).apply(1)
+      }
+    }
+
+    "not redirect and return 200 when a user is authenticated via a GU_U cookie" in new TestFixture {
+      val originalUrl = "https://profile.thegulocal.com/consents"
+      val request = Request(FakeRequest("GET", originalUrl), AnyContent())
+
+      when(authService.fullyAuthenticatedUser(any[RequestHeader])).thenReturn(Some(recentlyAuthedUser))
+      when(authService.consentCookieAuthenticatedUser(any[RequestHeader])).thenReturn(None)
+      when(client.me(any[Auth])).thenReturn(Future(Right(user)))
+      when(profileRedirectService.toConsentsRedirect(any[User], any[RequestHeader])).thenReturn(Future(NoRedirect))
+
+      val mockFunc = mock[Int => Result]
+      when(mockFunc.apply(1)) thenReturn mock[Result]
+      def callMock: AuthRequest[AnyContent] => Result = _ => mockFunc.apply(1)
+
+      val result = actions.consentsRedirectAction().apply(callMock)(request)
+      whenReady(result) { res =>
+        verify(mockFunc).apply(1)
+      }
+    }
+
+    "redirect to reauth when a user is authenticated via a GU_U cookie but not recently" in new TestFixture {
```

Comment:
We do want users to recently authenticate for consents? I guess we do but just checking.

Reply:
Yeah I am doubtful on this.  I think it's ok for now, but maybe we don't. Reason I left it like this is so that `AuthenticatedActions` didn't get any more complex.